sdk/src/client/api/block_builder/input_selection/mod.rs

```diff
@@ -270,6 +267,19 @@ impl InputSelection {
             }
         }
 
+        // If we're burning generated mana, set the capability flag.
+        if self.burn.as_ref().map_or(false, |b| b.generated_mana()) {
+            // Get the mana sums with generated mana to see whether there's a difference.
+            if !self
+                .transaction_capabilities
+                .has_capability(TransactionCapabilityFlag::BurnMana)
+                && input_mana < self.total_selected_mana(true)?
```

Comment:
I'm a bit confused by this. Why is this describing a burn if there is no output mana involved in the comparison?

Reply:
because we calculated `input_mana` without generated mana, so if there's a difference that means the output mana was also calculated without the generated mana, thus we have burned mana

Reply:
> because we calculated input_mana without generated mana

Are you sure about that?
`input_mana` is calculated from `self.total_selected_mana(None)` and None default to `include_generated = true`?

Reply:
yes, so if you are trying to burn mana, then my above statement is true, otherwise the amounts will always be the same and the flag will not be set